icu4j/main/classes/core/src/com/ibm/icu/util/MeasureUnit.java

```diff
@@ -695,7 +695,8 @@ public boolean equals(Object rhs) {
      */
     @Override
     public String toString() {
-        return type + "-" + subType;
+        String result = measureUnitImpl == null ? type + "-" + subType : measureUnitImpl.getIdentifier();
+        return result == null ? "" : result;
```

Comment:
Bringing in getIdentifier() logic. Else this returns "null-null", noticed in debug / unit test output.

Reply:
("Type" might be invalid for not-built-in-types, so I'm keeping "type" for built-ins and dropping it otherewise...)